docs make kubernetes_ca_cert optional on kubernetes auth (#25963)
hashicorp/vault: website/content/api-docs/auth/kubernetes.mdx

### Parameters

- `kubernetes_host` `(string: <required>)` - Host must be a host string, a host:port pair, or a URL to the base of the Kubernetes API server.

> - `kubernetes_ca_cert` `(string: "")` - Optional PEM encoded CA cert for use by the TLS client used to talk with the Kubernetes API.
>   NOTE: Every line must end with a newline: `\n`

Reviewer: ```suggestion
- `kubernetes_ca_cert` `(string: "")` - Optional PEM encoded CA cert. The TLS
  client can use the cert to talk with the Kubernetes API. **Every line in the
  cert file must end with a newline character (`\n`) rather than a carriage
  return (`\r`)**.
```
Added an assumed clarification re: newline vs carriage return, but it prompted a question: do we read CRLF as ending in a newline (`\r\n`)? Or must it be explicitly LF (`\n`)?
Style correction: active voice

Author: I've tested locally. Only it must be explicitly LF (\n).

Wondering if we should preserve the old NOTE's content and drop "rather than a carriage return (`\r`)" for simplicity.

  If not set, the local CA cert will be used if running in a Kubernetes pod.

>   If not set and `disable_local_ca_jwt` set to true, the system's trust store will be used.

Reviewer: ```suggestion
  If `kubernetes_ca_cert` is unset and `disable_local_ca_jwt` is `true`, Vault uses the Kubernetes trust store.
```
What is "the system" in "the system's trust store"? Is it Vault or Kubernetes? (I made an entirely uninformed guess for the sake of editing to provide an example of what I would suggest)

Author: "the system" here can be:
- the operating system if Vault is running on a machine
- the container operating system if Vault is running in a container

If Vault is running in a Kubernetes Pod, the `kubernetes_ca_cert` and
`token_reviewer_jwt` parameters will automatically default to the local CA cert
(`/var/run/secrets/kubernetes.io/serviceaccount/ca.crt`) and local service
account JWT (`/var/run/secrets/kubernetes.io/serviceaccount/token`).

> If this behavior is disabled by setting `disable_local_ca_jwt` to `true`,

Reviewer 2: Which behavior are you referring to?

Author: I'm referring to the behavior of `kubernetes_ca_cert` defaulting  to the local CA cert if Vault is running a Kubernetes pod from the previous paragraph https://github.com/hashicorp/vault/blob/77142a986adbcd634b77e985a4b34486a5f14824/website/content/api-docs/auth/kubernetes.mdx?plain=1#L67-L70

I can reiterate to make it clearer here. Let me know your thought.
```suggestion
If `kubernetes_ca_cert` defaulting to the local CA cert is disabled by setting `disable_local_ca_jwt` to `true`,
```

> If this behavior is disabled by setting `disable_local_ca_jwt` to `true`,

Reviewer: ```suggestion
If you override the default cert behavior by setting `disable_local_ca_jwt` to `true`,
```
Style correction: write in active voice, avoid "this" as a pronoun

> the plugin's TLS client will automatically default to use the system's trust store.

Reviewer 2: ```suggestion
the plugin's TLS client will automatically default to use the system's trust store for TLS certificate verification.
```

> the plugin's TLS client will automatically default to use the system's trust store for TLS certificate verification.

Reviewer: ```suggestion
the TLS client for the plugin automatically defaults to the Kubernetes trust store for TLS certificate verification.
```
Style correction: write in active voice, avoid possessives
